src/transformers/generation/utils.py

```diff
@@ -1449,12 +1449,39 @@ def _get_cache(
                     # models. May cause trobles with non-text modalities.
                     cache_dtype = self.get_output_embeddings().weight.dtype
 
+            def get_layer_device_mapping(execution_device: Optional[dict] = None):
+                if execution_device is None or len(execution_device) <= 1:
+                    return None
+                layer_device_mapping = {}
+                for layer in execution_device:
+                    for idx in range(self.config.num_hidden_layers):
+                        if f".{idx}." in f"{layer}.":
+                            layer_device_mapping[idx] = execution_device[layer]
+                            break
+                for idx in range(self.config.num_hidden_layers):
+                    if idx not in layer_device_mapping:
+                        raise RuntimeError(f"layer {idx} has not been mapped to a device.")
+                return layer_device_mapping
+
+            execution_device = None
+            # Taken from dispatch_model from accelerate.
+            # This is needed here if we don't want to make changes in accelerate in order to save execution_device
+            # For offloaded case, we need to get the execution device, not just the device where it is offloaded
+            if hasattr(self, "hf_device_map"):
+                main_device = main_device = [d for d in self.hf_device_map.values() if d not in ["cpu", "disk"]][0]
+                execution_device = {
```

Comment:
```suggestion
                execution_device_map = {
```
nit: I would expect `execution_device` to be a single device, suggesting a more precise name 🤗

Reply:
I will change that !